org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/IMarkdownCommentHelper.java

```diff
@@ -138,8 +138,9 @@ public void recordFenceChar(char previous, char next, boolean lineStarted) {
 		int required = this.insideFencedCodeBlock ? this.fenceLength : 3;
 		if (++this.fenceCharCount == required) {
 			this.insideFencedCodeBlock^=true;
-			this.fenceLength = this.fenceCharCount;
 		}
+		if (this.insideFencedCodeBlock && this.fenceCharCount >= this.fenceLength)
+			this.fenceLength = this.fenceCharCount;
```

Comment:
This part looks OK. 
I admit that the logic in this method is not very clear.
Do you want me to try an alternate design as an explicit state machine?
I'm thinking of states NONE, INCOMPLETE_OPEN, COMPLETE_OPEN, INSIDE, INCOMPLETE_CLOSE, where in particular COMPLETE_OPEN would be the state where a fence of 3 can still grow until a non-fence character is seen.

Reply:
If we don't come up with lot of issues in this regard, we can live with this, I guess.